plugins/fuse/server/server_fuse.py

```
@@ -0,0 +1,492 @@
+            read only.
+        :returns: a file descriptor.
+        """
+        resource = self._getPath(path)
+        if resource['model'] != 'file':
+            return super(ServerFuse, self).open(path, flags)
+        if flags & (os.O_APPEND | os.O_ASYNC | os.O_CREAT | os.O_DIRECTORY |
+                    os.O_EXCL | os.O_RDWR | os.O_TRUNC | os.O_WRONLY):
+            raise fuse.FuseOSError(errno.EROFS)
+        info = {
+            'path': path,
+            'handle': File().open(resource['document']),
+            'lock': threading.Lock(),
+        }
+        with self.openFilesLock:
+            fh = self.nextFH
+            self.nextFH += 1
+            self.openFiles[fh] = info
+        return fh
+
+    def release(self, path, fh):
+        """
+        Release an open file handle.
+
+        :param path: path within the fuse.
+        :param fh: an open file handle.
+        :returns: a file descriptor.
+        """
+        with self.openFilesLock:
+            if fh in self.openFiles:
+                with self.openFiles[fh]['lock']:
+                    if 'handle' in self.openFiles[fh]:
+                        self.openFiles[fh]['handle'].close()
+                        del self.openFiles[fh]['handle']
+                    del self.openFiles[fh]
+            else:
+                return super(ServerFuse, self).release(path, fh)
+        return 0
+
+    def destroy(self, path):
+        """
+        Handle shutdown of the FUSE.
+
+        :param path: always '/'.
+        """
+        events.trigger('server_fuse.destroy')
+        return super(ServerFuse, self).destroy(path)
+
+
+@atexit.register
+def unmountAll():
+    """
+    Unmount all mounted FUSE mounts.
+    """
+    for name in list(fuseMounts.keys()):
+        unmountServerFuse(name)
+
+
+def unmountServerFuse(name):
+    """
+    Unmount a mounted FUSE mount.  This may fail if there are open files on the
+    mount.
```

So does this mean I can call unmountServerFuse(...) and it can fail silently without any way of knowing that?

---

It won't be silent; it will report an error.  I'll have to refresh my memory on exactly what is reported, though.